components/event/CreateEvent/DeployEvent/DeployEventModal.tsx

```diff
@@ -1,24 +1,100 @@
 import { Backdrop, Box, Fade, Modal, Typography } from "@mui/material"
 import { Stack } from "@mui/system"
-import React from "react"
+import moment from "moment"
+import React, { Dispatch, SetStateAction, useContext, useState } from "react"
+import { useAccount } from "wagmi"
+import { useSmartContract } from "../../../../hooks/useSmartContract"
+import { Event } from "../../../../interfaces/event/event.interface"
 import ContainedButton from "../../../UI/button/ContainedButton"
 import FlatCard from "../../../UI/card/FlatCard"
+import Web3 from "web3"
+import { client } from "../../../../configs/axios/axiosConfig"
+import { UserContext } from "../../../../contexts/user/UserContext"
 
 type Props = {
   open: boolean
   handleClose: () => void
+  event: Event
+  setCurrentEvent: Dispatch<SetStateAction<Event | undefined>>
 }
 const style = {
   position: "absolute",
   top: "50%",
   left: "50%",
   transform: "translate(-50%, -50%)",
   width: 600,
-  bgcolor: "background.paper",
+  bgcolor: "#fff",
   boxShadow: 24
 }
 
-function DeployEventModal({ open, handleClose }: Props) {
+function DeployEventModal({
+  open,
+  handleClose,
+  event,
+  setCurrentEvent
+}: Props) {
+  const { address } = useAccount()
+  const { abi, bytecode } = useSmartContract()
+  const { user } = useContext(UserContext)
+
+  const getTicketSupply = (event: Event) => {
+    return (
+      event.ticketSupply.general +
+      event.ticketSupply.reservedSeat +
+      event.ticketSupply.vip
+    )
+  }
+
+  const smartContractArguments = [
+    address, // address _owner: wallet address of the smart contract owner
+    event._id, // string memory _eventID: id of the event document
+    event.name, // string memory _eventName: name of the event
+    event._id, // string memory _eventSymbol: symbol of the event
+    moment(event.doorTime).unix(), // _eventDoorTime
+    moment(event.startDate).unix(), // uint64 _eventStartDate,
+    moment(event.endDate).unix(), // uint64 _eventEndDate,
+    getTicketSupply(event), // TODO uint64 _ticketSupply maximum tickets allowed for this event
+    61000000000000, // TODO uint256 _initialTicketPrice: ticket price in wei unit
+    20, // TODO uint64 _maxPriceFactor: percentage of ticket price factor allowed
+    20 // TODO uint64 _transferFee: percentage of royalty fee collected by the event organizer when the ticket is resold
+  ]
+
+  const onCLickDeployContract = async (
+    account: `0x${string}`,
+    abi: any,
+    bytecode: any
+  ) => {
+    const web3 = new Web3(window.ethereum as any)
```

Comment:
not sure about this. But will window.eth become undefined when SSR?

Reply:
I think we can make it as a context or hook

Reply:
this won't be a problem for SSR because it is inside a handler function. It won't be triggered until a button press, which means it is only accessible after the page has been rendered on the client. Thus, window will be accessible always. The build has no problem as well.

Sure, I will extract this into a hook